rotkehlchen/db/upgrades/v28_v29.py

```diff
@@ -117,4 +117,9 @@ def upgrade_v28_to_v29(db: 'DBHandler') -> None:
     )
 
     _create_new_tables(db.conn)
+
+    # Rename uniswap_events table. Drop amm_events first if it was created at initialization
+    # of the db handler
+    cursor.execute('DROP TABLE IF EXISTS amm_events;')
```

Comment:
Why would it be? It should not be. There is nothing wrong with having it here per se, but not sure why you think it was needed.

Reply:
By running the tests It complained that the table already existed. I inspected the database and indeed was the case and assumed that since the creation statements are executed at initialization the table was created by default.

I checked now and indeed upgrades are executed before the create tables statements so I'm not sure what happened here
https://github.com/rotki/rotki/blob/develop/rotkehlchen/db/dbhandler.py#L316

Reply:
Which test complained? Could be somethign off in one of the old upgrade tests. Lots of things need to be patched there.

Reply:
Test for the migration during the migration :sweat: It said when trying to rename that the table already existed

Reply:
Which migration?

Did you think why this may be instead of just adding this check in the code? 

~~Perhaps it's the user DB that we have in the tests, probably added by you, that somehow due to your tests had it already?~~
It's not, I just checked.

You should find out why something is happening first.

Reply:
Yes I checked the test database before. Going to find the why :mag_right:

Reply:
Found why this table was being created. In this line we copy the v28 database https://github.com/rotki/rotki/blob/develop/rotkehlchen/tests/db/test_db_upgrades.py#L2212 and in next line we initialize the dbhandler with target v28. Since there is a new table in the list of tables to be created it gets created and commited to the database. Next when we initialize database for v29 the table was already created since we operate over the copied `rotkehlchen.db` file and the upgrade scripts complains because the table was created before (at initialization of the v28 database)

Reply:
Hmm yes we need to rethink this `use_prepared_db()` in the tests. Let's try to find out a way to improve this.